### PR TITLE
added data object to non json content type web calls

### PR DESCRIPTION
**Pull request by andysousa**
When making calls that are not application/json, set the first item in the data block to the body of the request.  This is especially important in OAuth requests.

The Native code was already doing this, this was only an issue on the web implementation.

Resolves https://github.com/capacitor-community/http/issues/12

### CODE REVIEW

File: src/web.ts [hunk 1]
```diff
@@ -57,10 +57,14 @@ export class HttpPluginWeb extends WebPlugin implements HttpPlugin {
 
     if (contentType.indexOf('application/json') === 0) {
       req['body'] = JSON.stringify(options.data);
-    } else if (contentType.indexOf('application/x-www-form-urlencoded') === 0) {
-    } else if (contentType.indexOf('multipart/form-data') === 0) {
-    }
-
+    } else if ((contentType.indexOf('application/x-www-form-urlencoded') === 0) ||
```
Comment (pixelbucket-dev) [inline src/web.ts:60]: You do not need the else here if you return in the first if early. Would make it more readable.

I realise it is not part of this MR, so would understand keeping it as it was.

Comment (pixelbucket-dev) [reply src/web.ts:60]: You can use _String.prototype.includes()_ then you don't have to handle indexes.

Comment (andysousa) [reply src/web.ts]: haha... why did i listen to you ;-). I will put it back

**Comment by mlynch** [comment]
Thanks for the PR!

**Comment by mlynch** [comment]
@all-contributors please add @andysousa for code

**Comment by mlynch** [comment]
@all-contributors please add @pixelbucket-dev for review

**Comment by dmastag** [comment]
@mlynch i don't see this one released inside the npm registry
https://www.npmjs.com/package/@capacitor-community/http

Is there something that needs to be done before we can have a new release?

**Comment by mghcs87** [comment]
Hi @mlynch and @andysousa, 
How can I use this solution in my project?
the version available in NPM is not updated :(

**Comment by andysousa** [comment]
@mghcs87 you can install GitHub code via npm
https://docs.npmjs.com/cli/install

**Comment by mghcs87** [comment]
Thanks @andysousa, it's working fine

**Pull request merged**